tritium-metrics/src/main/java/com/palantir/tritium/metrics/CacheMetrics.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics;
+
+import com.codahale.metrics.Gauge;
+import com.google.common.collect.ImmutableSortedMap;
+import com.palantir.tritium.metrics.registry.MetricName;
+import java.util.Map;
+import java.util.function.Function;
+
+public final class CacheMetrics {
```

Comment:
Ideally this wouldn't be public, but I suppose it must be in order to share with caffeine? If that's the case, can we add some javadoc describing that this isn't meant to be consumed, and perhaps rename to something including `Internal` (not a strong preference, but I'd look for a class called CacheMetrics if I wanted to add metrics to a cache)

Reply:
Renamed to `InternalCacheMetrics` and marked as `@Beta`